### DISCUSSION TITLE
catch exception in livecheck

### DISCUSSION THREAD

**Discussion by Yann-R** [Homebrew/discussions#4165] [Writing Formulae/Casks]
### Output of `brew config`

```shell
$ brew config
HOMEBREW_VERSION: 3.6.20-169-g856a347
ORIGIN: https://github.com/Homebrew/brew
HEAD: 856a347bfd0aa05ad644a864edf311ea1f7cbbf8
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9d2fca0f48c1a7966a141aad47456dfb92bc934c
Core tap last commit: 8 days ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/brew
HOMEBREW_REPOSITORY: /opt/brew/Homebrew
HOMEBREW_CELLAR: /opt/brew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.1QywUNaafY/org.xquartz:0
HOMEBREW_INSTALL_FROM_API: set
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.39.1 => /opt/brew/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6.2-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

### Output of `brew doctor`

```shell
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  xtail
  growly
  colorsvn
  contacts
  godep
  csv-fix
  szip

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  brew-pip
  gcc@6
  godep
  isl@0.18
  q
  rdesktop

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/ublio.h

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  unison

Warning: Some installed formulae are not readable:
  growly: wrong number of arguments (given 1, expected 0)

Warning: Your Homebrew's prefix is not /usr/local.

Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from MacHomebrew on Twitter.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

Hello,

While writing a personal formulae, I am facing a difficulty concerning the live check section.

For the concerned software
- I use the `strategy :page_match` from an url like this
```
    strategy :page_match do |page, regex|
      match = page.match(regex)
      "#{match[:num]},#{match[:tag]}"
    end
```
- but, unfortunately the url of the page can become unavailable (specifically for a new release) so future livechecks can raise an exception when the URL is no longer valid.

Since I can't find a more stable URL, is there any solution to catch the exception and return a correct result (like "new-release" or something else):
- that would at least avoid the failure
- and as a side-effect, that would be an acceptable workaround to notify an available update.

**Comment by gromgit** [comment]
> Since I can't find a more stable URL

If that's true, how does anyone find out what the new version is, and where to download it?

**Comment by Yann-R** [comment]
Good question.
The answer is that the "normal" (and human) way is to use the "search" about your device on the fabricant website (Dell) to reach the "new" download page (the result gives the most recent one).

Unfortunately, I did not find any way to use this "more general" and stable page of the search:
- it contains a link to the versioned (and new) page
- but with a display only visible after clicking and likely to be related to javascript that breaks all my matching effort
So, for now, I use in my formula the given webpage of the driver to check current version (that will disappear at next release)

E.g. 
a "stable" page (result from search) 
https://www.dell.com/support/home/en-us/product-support/product/dell-wd19-130w-dock/drivers
showing a version number when you click (Version 01.00.25, 01.00.09) but that I did not manage to reach from page match strategy
but containing an link to current driver page that I decided to use without perfect solution (but bound to current version, that will disappear with next release)
https://www.dell.com/support/home/en-us/drivers/driversdetails?driverid=5x3w3&oscode=wt64a&productcode=dell-wd19-130w-dock

I am aware that my idea is a kind of horrible workaround... Did not have better idea...

**Comment by Yann-R** [comment]
Hi @gromgit  

So, is there any solution to catch the exception and return a correct result when an URL for live check is no longer available?
Best regards.

**Reply by gromgit** [reply]
Not that I'm aware of, sorry.
